File: src/main/java/com/github/theword/queqiao/tool/response/ResponseEnum.java
package com.github.theword.queqiao.tool.response;

import com.github.theword.queqiao.tool.exception.response.ResponseException;

public enum ResponseEnum {
    SUCCESS("SUCCESS"), FAILED("FAILED");

    private final String value;

    ResponseEnum(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return value;
    }

    public static ResponseEnum fromString(String value) throws ResponseException {
        for (ResponseEnum response : ResponseEnum.values()) {
            if (response.value.equalsIgnoreCase(value)) {
                return response;
            }
        }
        throw ResponseException.unknownValue(value);
    }
}

File: src/main/java/com/github/theword/queqiao/tool/response/ResponseEnum.kt
package com.github.theword.queqiao.tool.response

enum class ResponseEnum(val value: String) {
    SUCCESS("success"),
    FAILED("failed");

    override fun toString(): String {
        return value
    }

    companion object {
        fun fromString(value: String): ResponseEnum {
            return entries.find { it.value.equals(value, ignoreCase = true) }
                ?: throw IllegalArgumentException("Unknown value: $value")
        }
    }
}
